### DISCUSSION TITLE
C++萬用標頭檔

### DISCUSSION THREAD

**Discussion by eric202501** [1101-GS4719-C/Python-Community-GS4719-C#1661] [General]
#include<bits/stdc++.h>
包含了大部分實用的函式庫，像是algorithm、string或各式STL容器
用了這個就不用分別引入了，畢竟像是考CPE的時候會緊張
但記得如果要用的話編譯器要選mingw

**Comment by v0ouo** [comment]
真是太驚人了!我學C++那麼久都沒有看過這個萬用標頭檔，昨天我考CPE時所有函式庫都要一個一個慢慢打，想必版主知道這個標頭檔，昨天寫CPE一定超過3題吧!

**Comment by 1chooo** [comment]
這確實非常好用，尤其在打競程的時候，不過用 Mac 的人好像不能使用，之前照著網路教學流程試了一堆還是乖乖 `#include` 了，不知道有沒有同學有沒有研究 Mac 使用萬用標頭檔 `#include <bits/stdc++.h>` 的方法

**Comment by Cerosop** [comment]
我也是，用過這個後就回不去了，我甚至不知道哪些東西是在哪個標頭擋的，這真的超方便!

**Comment by 12Jesse24** [comment]
太厲害了，最近剛好在學C++，每次遇到複雜一點的題目時，都要一個一個慢慢打，等下馬上去試試!

**Comment by poyen0806** [comment]
我個人還是喜歡乖乖一個一個標頭檔放上去
才能記得哪些函式是跟什麼有關的
而且Mac不能用#include <bits/stdc++.h>
所以不太方便...

啊不過這堂課是Python
所以還是提一下好了

Python好像沒有萬用的import的樣子
雖然C++的萬用標頭檔很方便
但還是建議各位一個一個import
以後才知道要用什麼功能時需要import什麼
還有虛擬環境需要pip install什麼

**Comment by SendaiHatsuki** [comment]
雖然`#include <bits/stdc++.h>`很方便，但是其實有一些缺點，像是編譯速度變慢還有占用記憶體空間，如果再進行大專案開發可能會有麻煩，我是習慣用PCH預編譯標頭。

**Comment by XiangLuen** [comment]
這東西只有 GCC 支援，在 MSVC 之類的環境下就爆炸，所以如果哪天寫專案或學校作業，還是手動 include 比較保險。而且它會把一堆根本用不到的東西一起 include 進來，對小型程式沒差，但要是大型專案，編譯時間可能會拖長。

**Comment by yue0903ZZ** [comment]
我是只有打競程的時候才用，超級方便。但在專案開發就不會了，因為浪費太多時間與記憶體了。
